ui/src/asm_cleanup.rs

```diff
@@ -89,9 +91,12 @@ pub fn filter_asm(block: &str) -> String {
             current_label = label_decl_cap.as_str(); 
         } else if DATA_REGEX.is_match(line) && current_label != "" { 
             line_info.push(Data(current_label));
-            // Skip the data directive, just add operands
+            // These will be checked for references to other labels later on
+            // Skip the data type, just capture its reference
             for label_ref_cap in LABEL_REF_REGEX.captures_iter(line).skip(1).filter_map(|cap| cap.get(1)) {
-                opcode_operands.insert(label_ref_cap.as_str());
+                // Capture unique labels referenced by data dirs in the current label
+                let label_set = data_dir_label_refs.entry(current_label).or_insert(HashSet::new());
```

Comment:
`.or_insert_with(HashSet::new)`

Reply:
What's the advantage of ```HashSet::new``` over ```HashSet::new()``` here?

Reply:
It avoids creating the `HashSet` when the value already exists.

In *this* case, it's not a big benefit as an empty `HashSet` doesn't allocate memory and is just a few pointers-worth of stack data. In other cases, it can have strong performance implications I (and clippy!) prefer it this way to be clear that the code only executes when needed.

Reply:
I think I understand, the function in ```or_insert``` is eagerly evaluated, but ```or_insert_with``` is lazy, right?

If that's correct, I might put a PR in for the [entry API docs](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_insert) to clarify this; I had to look at the "[or" method under Result](https://doc.rust-lang.org/std/result/enum.Result.html#method.or) for the implications to be spelled out.

Reply:
> the function in `or_insert` is eagerly evaluated, but or_insert_with is lazy, right?

That's right, except that `or_insert` doesn't take a function/closure, but only a value, so it *must* be evaluated.